packages/openapi-generator/src/parser/document.ts

```diff
@@ -25,8 +27,11 @@ export async function parseOpenApiDocument(
   const document = (await parse(clonedContent)) as OpenAPIV3.Document;
   const refs = await resolve(document);
   const originalFileName = removeFileExtension(basename(filePath));
+  const schemaInfo = parseSchemaInfo(document);
+  const schemaRefMapping = parseSchemaRefMapping(schemaInfo);
```

Comment:
I know you will not like this idea, because it it not 100% JavaScripty, but I am posting it anyway. So we could have a class like `ParsedSchemaManager`. This class knows all the parsed schemas and the mapping with names, refs  or other information we would need later potentially like some description in the docs. Then we pass this `ParsedSchemaManager` instance around. And call the `getOrginalName` `getReference` on this object later on.

Reply:
Actually, I expected a suggestion for a class from you, but not for this ;)
Let's discuss options in person, I would do this on a separate PR anyways.